tests/phpunit/CRM/MembershipExtras/Hook/Links/RecurringContributionTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+use CRM_MembershipExtras_Test_Fabricator_Contact as ContactFabricator;
+use CRM_MembershipExtras_Test_Fabricator_PaymentPlanOrder as PaymentPlanFabricator;
+use CRM_MembershipExtras_Test_Fabricator_MembershipType as MembershipTypeFabricator;
+use CRM_MembershipExtras_Hook_Links_RecurringContribution as RecurringContributionHook;
+use CRM_MembershipExtras_Test_Fabricator_RecurringContribution as RecurringContributionFabricator;
+
+/**
+ * Class CRM_MembershipExtras_Hook_Post_MembershipPaymentTest
+ *
+ * @group headless
+ */
+class CRM_MembershipExtras_Hook_Links_RecurringContributionTest extends BaseHeadlessTest {
+
+  /**
+   * Links to be altered by the hook.
+   *
+   * @var array
+   */
+  private $links = [];
+
+  public function setUp() {
+    $this->links = [
+      ['name' => 'Cancel', 'ref' => '', 'url' => ''],
+    ];
+
+    $this->setupNotManualPaymentProcessor();
+  }
+
+  public function testCancelLinkNotAlteredForContribution() {
```

Comment:
the test functions names are no longer accurate, it should be `testCancelLinkNotAlteredForNonManualPaymentPlan` and `testCancelLinkAlteredForManualPaymentPlan`